crates/relayer-framework-test/src/relayer_mock/contexts/chain.rs

```diff
@@ -0,0 +1,302 @@
+//! A HashMap<Height, ChainState> is used to represent the State of the
+//! chain.
+//! The current_state is the ChainState at all Heights, up to the latest Height.
+//! The consensus_states is a HashMap<ClientId, HashMap<Height, ChainState>>.
+//! This is used to check the state of a specific client, at a
+//! specific height.
+//! Usually the consensus_states would use the root hashes of a Merle Tree,
+//! but since the MockChain is used for testing and will have a small number
+//! of states, the whole state is used. This avoids needing to implement
+//! Merkle Trees and Proofs.
+
+use alloc::string::String;
+use eyre::eyre;
+use std::collections::hash_map::Entry;
+use std::sync::Mutex;
+use std::vec;
+use std::{collections::HashMap, sync::Arc};
+
+use crate::relayer_mock::base::error::Error;
+use crate::relayer_mock::base::types::aliases::{
+    ChainState, ChannelId, ClientId, ConsensusState, MockTimestamp, PortId, Sequence,
+};
+use crate::relayer_mock::base::types::events::Event;
+use crate::relayer_mock::base::types::message::Message as MockMessage;
+use crate::relayer_mock::base::types::runtime::MockRuntimeContext;
+use crate::relayer_mock::base::types::{
+    height::Height as MockHeight, packet::PacketKey, state::State,
+};
+use crate::relayer_mock::util::clock::MockClock;
+use crate::relayer_mock::util::mutex::MutexUtil;
+
+use ibc_relayer_framework::base::one_for_all::traits::runtime::OfaRuntimeContext;
+
+pub struct MockChainContext {
+    pub name: String,
+    pub past_chain_states: Arc<Mutex<HashMap<MockHeight, ChainState>>>,
+    pub current_state: Arc<Mutex<ChainState>>,
```

Comment:
I think what you need here is a `current_height` field and a `current_state` field, so that both operations are infallible inside the implementation.